charmcraft.yaml

```diff
@@ -126,41 +123,72 @@ bases:
 
 parts:
   charm:
-#    uncomment this if you add git+ dependencies in requirements.txt
-    # build-packages:
-    # - "git"
+    build-snaps:
+      - rustup
+      - astral-uv
+    override-build: |
+      rustup toolchain install stable
+      make generate-requirements
+      craftctl default
+    #    uncomment this if you add git+ dependencies in requirements.txt
+    #    build-packages:
+    #      - "git"
     charm-binary-python-packages:
       - "pydantic>=2"
       - "cryptography"
       - "jsonschema"
       - "opentelemetry-exporter-otlp-proto-http==1.21.0"
+    prime:
+      - -*.charm
+      - -spread.yaml
+      - -rockcraft.yaml
+      - -.venv
+      - -CONTRIBUTING.md
+      - -Makefile
+      - -pyproject.toml
+      - -README.md
+      - -requirements-dev.txt
+      - -scripts/
+      - -tests/
+      - -uv.lock
+      - -*.egg-info
```

Comment:
Can you explain the significance of this?  I'm not sure why we need it.

Reply:
it excludes these things from the packed artifact. We don't want to ship our tests in the charm